src/main/java/com/flab/demo/controller/ResponseEntityGroup.java

```diff
@@ -0,0 +1,20 @@
+package com.flab.demo.controller;
+
+import org.springframework.http.ResponseEntity;
+
+import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
+
+public enum ResponseEntityGroup {
+    OK_RESPONSE_ENTITY(ResponseEntity.ok().build()),
+    INTERNAL_SERVER_ERROR_RESPONSE_ENTITY(ResponseEntity.status(INTERNAL_SERVER_ERROR).build());
```

Comment:
이거를 상수가 아닌 enum으로 관리할 필요가 있을까요?
enum으로 관리하는 것의 장단점을 고민해보시면 좋을 것 같네요

Reply:
제가 알고 있는 Enum 의 장점으로는 

```
1. Type safety 와 value safety 를 제공해준다.
2. 별도로 불변을 위한 로직을 작성하지 않아도 직렬화나 reflection 으로 깨지지 않는 singleton 을 보장해준다.
3. java.lang.Enum 을 상속받기 때문에 해당 method 를 override 및 사용할 수 있다.
3. switch 문에서 value 로 사용할 수 있다.
4. value 가 아닌 name 으로 직렬화가 된다.
5. EnumSet 와 EnumMap 클래스들을 사용할 수 있다.
```

입니다.

단점으로는,

```
1. https://brunch.co.kr/@oemilk/94 와 
   https://medium.com/@kevalpatel2106/because-enum-requires-more-memory-and-in-many-java-applications-memory-is-crucial-thing-a6a3029fe1ef 
   내용처럼 Android 를 개발 시에는 주의해야된다.
2. Enum 을 상속받고 있기 때문에 추가적인 상속을 받지 못한다.
```

입니다. 

1번은 안드로이드는 시스템 메모리 부분에 별도로 할당하고 그 공간에 DEX 코드를 넣기 때문에 부하가 생길 수 있다는 점으로 알고 있는데, 안드로이드 개발이 아니기 때문에 문제되지 않는다고 생각하였습니다.

2번은 별도로 상속받을 이유가 없을 것이라고 생각하였고, 상수를 상속하는 것은 `anti-pattern` 으로 생각하기 때문에 문제되지 않는다고 생각하였습니다.


단순히 상수로 관리하는 것보다 많은 장점을 가질 수 있다고 생각했습니다.


혹시 제가 놓치고 있는 Enum 단점이나 상수로 했을 때의 이점이 있는 경우 지적해주시면 확인 후 수정하도록 하겠습니다. :mag:
피드백 감사합니다.! :memo: